backend/core/src/applications/services/applications.service.ts

```diff
@@ -66,14 +67,47 @@ export class ApplicationsService {
   async listPaginated(
     params: PaginatedApplicationListQueryParams
   ): Promise<Pagination<Application>> {
-    const qb = this._getQb(params)
-    const result = await paginate(qb, { limit: params.limit, page: params.page })
-    await Promise.all(
-      result.items.map(async (application) => {
-        await this.authorizeUserAction(this.req.user, application, authzActions.read)
+    if (!params.listingId) {
```

Comment:
what's the paginated case where we're fetching applications without a listing id?

Reply:
I'm honestly not sure, as it stands that endpoint doesn't require a listing id to be passed in
I can change it so we can drop that case but I wasn't sure if there was a reason for listing id to be not required

Reply:
@YazeedLoonat , let's make listingId a required parameter.